api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -119,9 +119,9 @@ message TcpProxy {
 
   // The idle timeout for connections managed by the TCP proxy filter. The idle timeout
   // is defined as the period in which there are no bytes sent or received on either
-  // the upstream or downstream connection. If not set, connections will never be closed
-  // by the TCP proxy due to being idle.
-  google.protobuf.Duration idle_timeout = 8 [(validate.rules).duration = {gt {}}];
+  // the upstream or downstream connection. If not set, the default idle timeout is 1 hour. If set
+  // to 0s, the timeout will be disabled.
```

Comment:
it occurs to me on not-my-PR it might be worth a word here and on the L7 timeout that if you do this you will almost surely have connection leaks.  Your call.